libarchive/archive_read_open_filename.c

```diff
@@ -556,7 +556,12 @@ static int64_t
 file_seek(struct archive *a, void *client_data, int64_t request, int whence)
 {
 	struct read_file_data *mine = (struct read_file_data *)client_data;
+#if defined(_WIN32) && !defined(__CYGWIN__)
+	/* We use _lseeki64() on Windows. */
```

Comment:
Is it appropriate to introduce a typedef `la_seek_t` to reduce the spread of preprocessor checks?

(Not blocking, just caught my eye)

Reply:
Alas, I'm on a trip and don't have my headers handy ☹️

I do know we had a define for _32-bit `time_t`_, but I think it's existence may the exception and not the rule...

Reply:
I have implemented your `la_seek_t` idea. Since I'm not that great with names, I'm open for better ideas here.

We need a way to `fstat`/`stat` a file on Windows and work with a `struct stat` which uses `off_t`. So, we need some wrapper here. I have introduced `archive_platform_stat.h` for this, which uses its own stat struct on Windows and otherwise uses the system calls available.

If someone has a good idea for a name, I'll adjust the code accordingly!

Reply:
Renamed to `la_fsize_*` since these wrappers exist now to handle file sizes in general. Also added a comment in the new header file, explaining that these wrappers are only needed if `st_size` is used.